packages/devtools/src/sourceMap/loadOriginalSourceLoc.ts

```diff
@@ -0,0 +1,178 @@
+          // store computed sourceMapJSON
+          if (sourceMapNum > 1) {
+            sourceMapJSONs.set(`${source}:${line}:${column}`, sourceMapJSON);
+          } else {
+            if (sourceMapRegex.exec(runtimeSourceCode) === null) {
+              sourceMapJSONs.set(source, sourceMapJSON);
+            } else {
+              sourceMapJSONs.set(`${source}:${line}:${column}`, sourceMapJSON);
+            }
+          }
+
+          return getOriginalPosition(sourceMapJSON, sourceLoc);
+        }
+      } catch (error) {
+        // We've likely encountered a string in the source code that looks like a source map but isn't.
+        // Maybe the source code contains a "sourceMappingURL" comment or something similar.
+        // In either case, let's skip this and keep looking.
+      }
+    } else {
+      externalSourceMapURLs.push(sourceMappingURL);
+    }
+
+    sourceMappingURLMatch = sourceMapRegex.exec(runtimeSourceCode);
+  }
+
+  // No inline sourcemap found. Check if there's external source map
+  let sourceMappingURL = externalSourceMapURLs.pop();
+  if (!sourceMappingURL) {
+    throw new Error(
+      `[Griffel devtools] extractAndLoadSourceMapJSON() encountered string "sourceMappingURL" in file ${source} that does not contain a valid source map`,
+    );
+  }
+
+  if (externalSourceMapURLs.length > 0) {
+    console.warn(
+      `[Griffel devtools] extractAndLoadSourceMapJSON() More than one external source map detected in the file ${source}:`,
+    );
+    externalSourceMapURLs.forEach(sourceMappingURL => {
+      console.warn(`[Griffel devtools] extractAndLoadSourceMapJSON() skipping source map "${sourceMappingURL}"`);
+    });
+  }
+
+  if (!sourceMappingURL.startsWith('http') && !sourceMappingURL.startsWith('/')) {
+    const lastSlashIdx = source.lastIndexOf('/');
+    if (lastSlashIdx !== -1) {
+      const baseURL = source.slice(0, lastSlashIdx);
+      sourceMappingURL = `${baseURL}/${sourceMappingURL}`;
+    }
+  }
+
+  const sourceMapContent = await fetchFiles(sourceMappingURL);
+  const sourceMapJSON = JSON.parse(sourceMapContent);
+  sourceMapJSONs.set(source, sourceMapJSON);
+
+  return getOriginalPosition(sourceMapJSON, sourceLoc);
+}
+
+async function fetchFiles(url: string): Promise<string> {
+  try {
+    const response = await fetch(url);
+    if (response.ok) {
+      return response.text();
+    }
+    throw new Error(`[Griffel devtools] fetchRuntimeSource() bad response fetching ${url}: ${response.status}`);
+  } catch (error) {
+    // try to get url content from chrome resources api
+    const resource = (await resources).find(resource => resource.url === url);
+    if (resource) {
+      return new Promise(resolve => resource.getContent(content => resolve(content)));
+    }
+    throw new Error(`[Griffel devtools] fetchFiles() unable to fetch ${url}: ${error}`);
+  }
+}
+
+function decodeBase64String(encoded: string): string {
+  if (typeof atob === 'function') {
```

Comment:
I think this is not a problem either. I copied this part directly from react devtools https://github.com/facebook/react/blob/b66936ece7d9ad41a33e077933c9af0bda8bff87/packages/react-devtools-shared/src/hooks/parseHookNames/loadSourceAndMetadata.js#L121

Now to think about it, maybe it's because code in react devtools is not only running on the client side. I'm going to simplify this and keep only `atob`